### PR TITLE
Block context switch when uploading image

### PR DESCRIPTION
**Pull request by ddc22**
#### Changes proposed in this Pull Request

* When uploading images the user can context switch while an async image upload is happening leading to a dirty unpredictable state. This change aims to make image upload a blocking action for the user.

#### Testing instructions
* Go through the DIFM flow, checkout and reach the website-content state
* Upload an image anywhere and make sure switching across form sections is blocked
* Clicking on other images inside sections should be allowed

### CODE REVIEW

File: client/signup/steps/new-or-existing-site/index.tsx [hunk 1]
```diff
@@ -18,8 +21,10 @@ export default function NewOrExistingSiteStep( props: Props ): React.ReactNode {
 	const dispatch = useDispatch();
 	const translate = useTranslate();
 
-	const headerText = translate( 'Choose where you want us to build your site.' );
-	const subHeaderText = '';
+	const headerText = translate( 'Do it For Me' );
```
Comment (a8ci18n) [inline client/signup/steps/new-or-existing-site/index.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **17** times:
`translate( 'Do It For Me (Lite)' )` *ES Score: 7*

File: client/state/signup/steps/website-content/schema.ts [hunk 2]
```diff
@@ -55,9 +55,11 @@ export type WebsiteContent = Array< PageData >;
 export interface WebsiteContentCollection {
 	currentIndex: number;
 	websiteContent: WebsiteContent;
+	imageUploadStates: Record< string, Record< string, string > >;
```
Comment (aneeshd16) [inline client/state/signup/steps/website-content/schema.ts:61]: `imageUploadStates` should also be a part of the `schema` above.

File: client/state/signup/steps/website-content/reducer.ts [hunk 3]
```diff
@@ -37,10 +45,24 @@ export default withSchemaValidation(
 					currentIndex: action.payload,
 				};
 
-			case SIGNUP_STEPS_WEBSITE_CONTENT_IMAGE_UPLOADED: {
+			case SIGNUP_STEPS_WEBSITE_CONTENT_IMAGE_UPLOAD_STARTED: {
 				const { payload } = action;
 
-				const pageIndex = state.websiteContent.findIndex( ( page ) => page.id === payload.id );
+				return {
```
Comment (aneeshd16) [inline client/state/signup/steps/website-content/reducer.ts:51]: I was wondering if it's required to store the upload state for every `pageId`/`mediaIndex`? We need to block navigation if any image upload is in progress, so a simple boolean flag can also work here. Similar to this reducer:
https://github.com/Automattic/wp-calypso/blob/fff000dd11e0a0fb37d9b285ae872c5ea9ca7fc5/client/state/sites/reducer.js#L282-L293

**Review by aneeshd16 — COMMENTED** [review]
Works well! I have added some code comments.
Also, while noticed that the arrow on the "Next" button is visible after a slight delay. This lag is not there in production.

https://user-images.githubusercontent.com/5436027/155305109-faede72f-bcde-4772-a5d3-edc4fb9a6444.mp4

File: client/state/signup/steps/website-content/reducer.ts (hunk 3 above)
Comment (ddc22) [reply client/state/signup/steps/website-content/reducer.ts:51]: Actually the main motivation for this is to allow multiple images to be upload within a section. Also if we want to allow switching between sections, or show an upload progress indicated in a section while images are being uploaded as a UX enhancement having the section vise state will be beneficial.

**Comment by ddc22** [comment]
> Also, while noticed that the arrow on the "Next" button is visible after a slight delay. This lag is not there in production.

Might be a grid icon loading related issue 🤔 

Also I have addressed your comments @aneeshd16

**Review by aneeshd16 — APPROVED** [review]

**Pull request merged**